src/main/java/seedu/modtracker/ModuleList.java

```diff
@@ -2,37 +2,121 @@
-        modList.add(currentModule);
-        System.out.println(currentModule + " is added" + System.lineSeparator());
+        String[] modInfo = input.split(" ", 2);
+        modInfo[1] = modInfo[1].toUpperCase(); //convert module code to upper case
+        if (!checkModuleIfExist(modInfo[1])) {
+            Module currentModule = new Module(modInfo[1]);
+            modList.add(currentModule);
+            ui.printAdd(currentModule);
+        } else {
+            ui.printExist(modInfo[1]);
+        }
     }
 
+    /**
+     * Creates a module and adds the module with expected time to the
+     * list of modules if module does not exist.
+     * If module already exist, update expected time based on user input
+     *
+     * @param input module code and expected time typed in by user
+     */
     public void addExp(String input) {
         String[] modInfo = input.split(" ", 3);
+        modInfo[1] = modInfo[1].toUpperCase();  //convert module code to upper case
         Module currentMod = new Module(modInfo[1], modInfo[2]);
-        modList.add(currentMod);
-        System.out.println(currentMod + " is added" + System.lineSeparator());
+        if (!checkModuleIfExist(modInfo[1])) {
+            modList.add(currentMod);
+        } else {
+            int index = modList.indexOf(currentMod);
+            int expectedTime = Integer.parseInt(modInfo[2]);
+            modList.get(index).expected = expectedTime;
+        }
+        ui.printAdd(currentMod);
+    }
+
+    /**
+     * Deletes the module if module exists.
+     *
+     * @param input module code typed in by user
+     */
+    public void deleteMod(String input) {
+        String[] modInfo = input.split(" ", 2);
+        modInfo[1] = modInfo[1].toUpperCase(); //convert module code to upper case
+        if (checkModuleIfExist(modInfo[1])) {
+            Module inputMod = new Module(modInfo[1]);
+            modList.remove(inputMod);
+            ui.printDelete(modInfo[1]);
+        } else {
+            ui.printNotExist(modInfo[1]);
+        }
+    }
+
+    /**
+     * Deletes the expected time of the module if module exists.
+     *
+     * @param input module code and expected time typed in by user
+     */
+    public void deleteExp(String input) {
+        String[] modInfo = input.split(" ", 2);
+        modInfo[1] = modInfo[1].toUpperCase();  //convert module code to upper case
+        if (checkModuleIfExist(modInfo[1])) {
+            Module inputMod = new Module(modInfo[1]);
+            int index = modList.indexOf(inputMod);
+            modList.get(index).expected = -1;
+            ui.printDeleteExp(modInfo[1]);
+        } else {
+            ui.printNotExist(modInfo[1]);
+        }
+    }
+
+    /**
+     * Checks if the module exists in the list of modules.
+     *
+     * @param input module code typed in by user
+     * @return the existence of module in the list of modules.
+     */
+    public boolean checkModuleIfExist(String input) {
```

Comment:
Is "checkIfModuleExist" a better name?

Reply:
Yes, the function name is renamed to checkIfModuleExist.